src/Microsoft.AspNet.Mvc.ModelBinding/Validation/DataMemberModelValidatorProvider.cs

```diff
@@ -6,49 +6,40 @@
 using System.Linq;
 using System.Reflection;
 using System.Runtime.Serialization;
+using Microsoft.AspNet.Mvc.ModelBinding.Metadata;
 
-namespace Microsoft.AspNet.Mvc.ModelBinding
+namespace Microsoft.AspNet.Mvc.ModelBinding.Validation
 {
     /// <summary>
     /// This <see cref="IModelValidatorProvider"/> provides a required ModelValidator for members marked
     /// as <c>[DataMember(IsRequired=true)]</c>.
     /// </summary>
-    public class DataMemberModelValidatorProvider : AssociatedValidatorProvider
+    public class DataMemberModelValidatorProvider : IModelValidatorProvider
     {
-        protected override IEnumerable<IModelValidator> GetValidators(ModelMetadata metadata,
-                                                                      IEnumerable<object> attributes)
+        public void GetValidators(ModelValidatorProviderContext context)
         {
             // Types cannot be required; only properties can
-            if (metadata.ContainerType == null || string.IsNullOrEmpty(metadata.PropertyName))
+            if (context.ModelMetadata.MetadataKind != ModelMetadataKind.Property)
```

Comment:
What do you mean more specific? `metadata.ContainerType == null` and `context.ModelMetadata.MetadataKind != ModelMetadataKind.Property` are equivalent.